apps/semantic-cache/src/analytics.ts

```diff
@@ -0,0 +1,30 @@
+import { Tinybird } from "@chronark/zod-bird";
+import { z } from "zod";
+
+export class Analytics {
+  public readonly client: Tinybird;
+
+  constructor(opts: {
+    tinybirdToken: string;
+  }) {
+    this.client = new Tinybird({ token: opts.tinybirdToken });
+  }
+
+  public get ingestLogs() {
+    return this.client.buildIngestEndpoint({
+      datasource: "semantic_cache__v3",
+      event: z.object({
+        timestamp: z.string(),
+        model: z.string(),
+        stream: z.boolean(),
+        query: z.string().optional(),
+        vector: z.array(z.number()).optional(),
+        response: z.string().optional(),
+        cache: z.boolean().optional(),
+        timing: z.number().optional(),
```

Comment:
how come all of these are optional?

it feels like they would be present on every request

Reply:
It's because the event is initialized with some of these properties, with the remaining properties added in the waitUntil. I split it into two types to handle that